node/tasks.py

```diff
@@ -101,50 +102,57 @@ def send_app_batches(app_name: str) -> dict[str, Any]:
 
     check_finalization()
     return response
-        
 
 
 def sync_with_sequencer(
-    app_name: str, initialized_batches: dict[str, Any], sequencer_response: dict[str, Any]
+        app_name: str, initialized_batches: dict[str, Any], sequencer_response: dict[str, Any]
 ) -> dict[str, Any]:
     """Sync batches with the sequencer."""
     zdb.upsert_sequenced_batches(app_name=app_name, batches_data=sequencer_response["batches"])
     last_locked_index: str = zdb.apps[app_name]["last_locked_batch"].get("index", 0)
-    if sequencer_response["locked"]["index"] > last_locked_index:
-        if is_sync_point_signature_verified(
+    last_finalized_index: str = zdb.apps[app_name]["last_finalized_batch"].get("index", 0)
+
+    if (sequencer_response["locked"]["index"] != 0 and
+            sequencer_response["locked"]["index"] <= last_locked_index):
+        zlogger.error("Invalid locking signature received from sequencer")
```

Comment:
it's better to log "out of order locking signature ..."

Reply:
also in the previous implementation, if such a thing happens `update_finalized_batches` will not be called anymore, but it seems to be changed in this update, why?

Reply:
> it's better to log "out of order locking signature ..."

When invalid signature is detected it is not clear whether it is about out of order or other reasons